From 913758b4a87deffc4a723b948c4aa855dc233930 Mon Sep 17 00:00:00 2001
From: Vladimir Yaichkin <vyaichkin@rzn.dlink.ru>
Date: Mon, 17 Aug 2020 10:49:55 +0300
Subject: strongswan: add reread secret file to unlock sem
Patch-from-hash: aff0e04810cefd30f3b489990d3fb96d07619951
Patch-repo: strongswan
Patch-branch: master
Patch-version: 5

Добавлена отправка завершения загрузки информации в
secret-файл.

Signed-off-by: Vladimir Yaichkin <vyaichkin@rzn.dlink.ru>
---
 src/libcharon/plugins/stroke/stroke_cred.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/libcharon/plugins/stroke/stroke_cred.c b/src/libcharon/plugins/stroke/stroke_cred.c
index 7fc9565..e60982c 100644
--- a/src/libcharon/plugins/stroke/stroke_cred.c
+++ b/src/libcharon/plugins/stroke/stroke_cred.c
@@ -1404,6 +1404,9 @@ METHOD(stroke_cred_t, reread, void,
	{
		DBG1(DBG_CFG, "rereading secrets");
		load_secrets(this, NULL, this->secrets_file, 0, prompt);
+#ifdef DLINK_SOFTWARE
+		dlink_service_update(json_pack("{s:b}", "Reread", 1));
+#endif /* DLINK_SOFTWARE */
	}
	if (msg->reread.flags & REREAD_CACERTS)
	{
--
2.20.1
